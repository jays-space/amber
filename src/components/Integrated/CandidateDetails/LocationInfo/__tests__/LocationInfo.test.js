import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { LocationInfo } from "..";

describe("<LocationInfo />", () => {
  const mockLocationData = {
    street: {
      number: 456,
      name: "sicamore street",
    },
    city: "rivonia",
    state: "johannesburg",
    country: "south africa",
    postcode: 1234,
    coordinates: {
      latitude: "-54.4721",
      longitude: "30.6940",
    },
    timezone: {
      offset: "+2:00",
      description: "cape town, harare",
    },
  };

  it("should render correctly", () => {
    render(<LocationInfo location={mockLocationData} />);

    const locationInfoComponent = screen.getByTitle("candidate-address");
    expect(locationInfoComponent).toBeInTheDocument();
    expect(locationInfoComponent).toMatchSnapshot();
  });

  it("should render appropriate heading", () => {
    render(<LocationInfo location={mockLocationData} />);

    const headingElement = screen.getByRole("heading", { name: "address" });
    expect(headingElement).toBeInTheDocument();
  });

  it("should render street address details", () => {
    render(<LocationInfo location={mockLocationData} />);

    const addressComponent = screen.getByTestId("street address-details");
    expect(addressComponent).toBeInTheDocument();
  });

  it("should render city details", () => {
    render(<LocationInfo location={mockLocationData} />);

    const cityComponent = screen.getByTestId("city-details");
    expect(cityComponent).toBeInTheDocument();
  });

  it("should render state details", () => {
    render(<LocationInfo location={mockLocationData} />);

    const stateComponent = screen.getByTestId("state-details");
    expect(stateComponent).toBeInTheDocument();
  });

  it("should render country details", () => {
    render(<LocationInfo location={mockLocationData} />);

    const countryComponent = screen.getByTestId("country-details");
    expect(countryComponent).toBeInTheDocument();
  });

  it("should render postcode details", () => {
    render(<LocationInfo location={mockLocationData} />);

    const postcodeComponent = screen.getByTestId("postcode-details");
    expect(postcodeComponent).toBeInTheDocument();
  });

  it("should render timezone details", () => {
    render(<LocationInfo location={mockLocationData} />);

    const timezoneComponent = screen.getByTestId("timezone-details");
    expect(timezoneComponent).toBeInTheDocument();
  });

  it("should render the map", () => {
    render(<LocationInfo location={mockLocationData} />);

    const imageElement = screen.getByRole("img");
    expect(imageElement).toBeInTheDocument();
  });
});
