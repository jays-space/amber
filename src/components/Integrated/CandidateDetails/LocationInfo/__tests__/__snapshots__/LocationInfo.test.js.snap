// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationInfo /> should render correctly 1`] = `
<div
  class="mt-10 lg:mt-0"
  title="candidate-address"
>
  <h4
    class="mb-6 md:mb-2 text-slate-700 capitalize text-xl font-bold"
  >
    address
  </h4>
  <div
    class="grid md:grid-cols-2 gap-6 md:border rounded-md"
  >
    <div
      class="md:p-4 grid grid-cols-1 md:grid-cols-2 lg:grid-cols-1 gap-6"
    >
      <div
        class="flex flex-col "
        data-testid="street address-details"
      >
        <p
          class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
          data-testid="paragraph"
        >
          street address
        </p>
        <span
          class="flex"
        >
          <p
            class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
            data-testid="paragraph"
          >
            456,
          </p>
          <p
            class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
            data-testid="paragraph"
          >
            sicamore street
          </p>
        </span>
      </div>
      <div
        class="flex flex-col "
        data-testid="city-details"
      >
        <p
          class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
          data-testid="paragraph"
        >
          city
        </p>
        <span
          class="flex"
        >
          <p
            class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
            data-testid="paragraph"
          >
            rivonia
          </p>
        </span>
      </div>
      <div
        class="flex flex-col "
        data-testid="state-details"
      >
        <p
          class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
          data-testid="paragraph"
        >
          state
        </p>
        <span
          class="flex"
        >
          <p
            class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
            data-testid="paragraph"
          >
            johannesburg
          </p>
        </span>
      </div>
      <div
        class="flex flex-col "
        data-testid="country-details"
      >
        <p
          class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
          data-testid="paragraph"
        >
          country
        </p>
        <span
          class="flex"
        >
          <p
            class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
            data-testid="paragraph"
          >
            south africa
          </p>
        </span>
      </div>
      <div
        class="flex flex-col "
        data-testid="postcode-details"
      >
        <p
          class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
          data-testid="paragraph"
        >
          postcode
        </p>
        <span
          class="flex"
        >
          <p
            class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
            data-testid="paragraph"
          >
            1234
          </p>
        </span>
      </div>
      <div
        class="flex flex-col "
        data-testid="timezone-details"
      >
        <p
          class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
          data-testid="paragraph"
        >
          timezone
        </p>
        <span
          class="flex"
        >
          <p
            class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
            data-testid="paragraph"
          >
            cape town, harare
          </p>
          <p
            class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
            data-testid="paragraph"
          >
            (+2:00)
          </p>
        </span>
      </div>
    </div>
    <div
      class="overflow-hidden"
    >
      <img
        alt="map of coordinates"
        class="h-full object-cover object-center rounded-md"
        src="/map.png"
      />
    </div>
  </div>
</div>
`;
